### DISCUSSION TITLE
atomic layout export the component for reuse

### DISCUSSION THREAD

**Discussion by njavilas** [preactjs/preact#3650] [Q&A]
good in my project I am using atomic design where it allows me to export the component to reuse it without being of the es or udm type, how could I export a component made with preact to install it in another project only with build code

I am looking for a way to adapt some suggestion? I tried to copy vite configuration to build but it doesn't take me because the html has to exist

```#tsconfig.json
{
  "compilerOptions": {
    "target": "ES6",
    "module": "commonjs",
    "lib": [
      "dom",
      "es6"
    ],
    "jsx": "react",
    "declaration": true,
    "outDir": "lib",
    "rootDir": "src",
    "removeComments": true,
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true
  },
  "exclude": [
    "node_modules",
  ]
}
```

```ts src/index.ts
import React from "react";
import styled from "styled-components";

const Container = styled.div`

`

interface params {
  
}

const App = (params: params): JSX.Element => {

  
    return (
        <Container>


        </Container>
    );

};

App.defaultProps = {
   
};

export default App;
```
code build
![image](https://user-images.githubusercontent.com/47610140/183230428-9e8d782e-5065-4898-ae97-e5a063fe205b.png)

**Comment by marvinhagemeister** [comment]
What kind of error are you getting? Did you follow [the instructions in the vite docs](https://vitejs.dev/guide/build.html#library-mode)?
